malloc/malloc.c

```diff
@@ -4,43 +4,83 @@
 #include "lib.h"
 
 /*
-    Duplicates the input string by dynamically allocating memory for 
+    Duplicates the input string by dynamically allocating memory for
     the duplicate string using `malloc` and then copying the string
     into the allocated memory. Returns a pointer to the allocated memory.
     You may want to use the string_length function to figure out the
     length of the input string.
 */
 char *string_dup(char *src)
 {
+    int sLen = string_length(src);
+    // Yes, chars are 1 byte. Yes, I'm still being explicit for readability sake.
+    char *dupe = malloc(sLen * sizeof(char));
```

Comment:
it's actually great practice though. no need to assume that every single programmer going through your code will know everything, or understand that size 1 byte is meant for chars.